core/tee/tee_ree_fs.c

```diff
@@ -588,18 +590,51 @@ static void close_dirh(struct tee_fs_dirfile_dirh **dirh)
 #else /*!CFG_REE_FS_INTEGRITY_RPMB*/
 static TEE_Result open_dirh(struct tee_fs_dirfile_dirh **dirh)
 {
-	TEE_Result res;
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t min_counter = 0;
+
+	res = nv_counter_get_ree_fs(&min_counter);
+	if (res) {
+		static bool once;
+
+		if (res != TEE_ERROR_NOT_IMPLEMENTED ||
+		    !IS_ENABLED(CFG_WARN_INSECURE))
+			return res;
 
-	res = tee_fs_dirfile_open(false, NULL, &ree_dirf_ops, dirh);
+		if (!once) {
+			IMSG("WARNING (insecure configuration): Failed to get monotonic counter for REE FS, using 0");
+			once = true;
+		}
+		min_counter = 0;
+	}
+	res = tee_fs_dirfile_open(false, NULL, min_counter, &ree_dirf_ops,
+				  dirh);
 	if (res == TEE_ERROR_ITEM_NOT_FOUND)
```

Comment:
Well spotted. Yes, that makes sense. Perhaps using the same logic as with the hash in RPMB above, that is, `IS_ENABLED(CFG_REE_FS_ALLOW_RESET)`. I'll add something.

I guess it would be quite useful to be able to reset secure storage via a PTA too, but that's not in scope here.